astropy/io/fits/hdu/hdulist.py

```diff
@@ -184,6 +185,14 @@ def fitsopen(
 
         .. versionadded:: 5.2
 
+    decompress_in_memory : bool, optional
+        By default blocks are decompressed on the fly, when reading them.  This
```

Comment:
I think the use of 'blocks' isn't necessarily going to be clear to the average user. Can we instead say something like 'By default, files are decompressed progressively depending on what data is needed' (not attached to this wording, but just not using blocks - which to me makes me think of tiles in tiled image compression)